### PR TITLE
Added support to dynamicDefaults for the new useDefaults option empty

### PR DESCRIPTION
**Pull request by B-Teague**
I fixed a few typos in the readme, added mocha global variable names to the eslint configuration, and fixed all the eslint errors in the dynamicDefaults source code.

Let me know if there are any additional assertions or test cases you want me to add.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -619,19 +619,18 @@ var schema = {
   type: 'object',
   dynamicDefaults: {
     ts: 'datetime',
-    r: { func: 'randomint', max: 100 },
-    id: { func: 'seq', name: 'id' }
+    r: { func: 'randomint', args: { max: 100 } },
+    id: { func: 'seq', args: { name: 'id' } }
```
[the remark this answers is not in the document]
Comment (B-Teague) [reply README.md]: Fixed.

File: keywords/dynamicDefaults.js [hunk 2]
```diff
@@ -35,8 +35,12 @@ module.exports = function defFunc(ajv) {
               : noop;
 
       function assignDefaults(data) {
-        for (var prop in schema)
-          if (data[prop] === undefined) data[prop] = funcs[prop]();
+        for (var prop in schema){
+          if (data[prop] === undefined
+            || it.opts.useDefaults == 'empty'
+            && (data[prop] === null || data[prop] === ''))
+            data[prop] = funcs[prop]();
```
[the remark this answers is not in the document]
Comment (B-Teague) [reply keywords/dynamicDefaults.js]: Added explicit parentheses.

File: spec/dynamicDefaults.spec.js [hunk 3]
```diff
@@ -151,7 +150,7 @@ describe('keyword "dynamicDefaults"', function() {
 
       var data1 = {};
       test(data1);
-      assert.notEqual(data.id, data1.id);
+      assert.notEqual(data0.id, data1.id);
```
[the remark this answers is not in the document]
Comment (B-Teague) [reply spec/dynamicDefaults.spec.js]: Removed as requested.  eslint will continue to give the warning, "'data' is already declared in the upper scope."

File: spec/dynamicDefaults.spec.js [hunk 4]
```diff
@@ -164,4 +163,114 @@ describe('keyword "dynamicDefaults"', function() {
...
+        validate(data1).should.equal(true);
+        data.ts.should.not.equal(data1.ts);
+        data1.r.should.be.a('number');
+        //data.r and data1.r could match, but unlikely
+        data.id.should.not.equal(data1.id);
+        done();
+      }, 1000);
+    }
+  });
+});
```
[the remark this answers is not in the document]
Comment (B-Teague) [reply spec/dynamicDefaults.spec.js]: Added back missing EOL

File: .eslintrc.yml [hunk 5]
```diff
@@ -1,5 +1,6 @@
 env:
   node: true
+  mocha: true
```
[the remark this answers is not in the document]
Comment (B-Teague) [reply .eslintrc.yml]: Removed mocha as requested.  eslint will continue to give the error, "'describe' is not defined" and "'it' is not defined".

File: README.md [hunk 6]
```diff
@@ -595,7 +595,7 @@ __Please note__: the current implementation is BETA. It does not allow using rel
 
 This keyword allows to assign dynamic defaults to properties, such as timestamps, unique IDs etc.
 
-This keyword only works if `useDefaults` options is used and not inside `anyOf` keywrods etc., in the same way as [default keyword treated by Ajv](https://github.com/epoberezkin/ajv#assigning-defaults).
+This keyword only works if `useDefaults` options is used and not inside `anyOf` keywords etc., in the same way as [default keyword treated by Ajv](https://github.com/epoberezkin/ajv#assigning-defaults).
```
[the remark this answers is not in the document]
Comment (B-Teague) [reply README.md]: Removed extra space.

**Comment by B-Teague** [comment]
Okay, I'm a little new to the review process on github, but I believe I have resolved all your requests with the new commit.  Let me know if I missed anything.  Thanks!

File: spec/dynamicDefaults.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (epoberezkin) [reply spec/dynamicDefaults.spec.js]: thanks - that's a separate issue - can be a separate PR

File: .eslintrc.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (epoberezkin) [reply .eslintrc.yml]: same - separate problem - can be a separate PR - created issue #81 for eslint warnings.

**Pull request merged**